activerecord/test/cases/inheritance_test.rb

```diff
@@ -208,6 +208,16 @@ def test_new_with_complex_inheritance
     assert_nothing_raised { Client.new(type: 'VerySpecialClient') }
   end
 
+  def test_new_without_storing_full_sti_class
+    old = ActiveRecord::Base.store_full_sti_class
```

Comment:
I pushed a little refactoring containing a helper to assign the configuration (https://github.com/rails/rails/commit/d27974d3fb38918d323dcce3ae439ef68b369b72). Could you update this test case to use the helper as well?